<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TransformationWidget</class>
 <widget class="QDockWidget" name="TransformationWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>328</width>
    <height>413</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Affine Transformation</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Transformation Matrix</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="2" column="2">
            <widget class="QLabel" name="matrix_22">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="matrix_20">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLabel" name="matrix_03">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="matrix_21">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="matrix_12">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="matrix_00">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="matrix_01">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QLabel" name="matrix_13">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QLabel" name="matrix_23">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="matrix_11">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="matrix_10">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="matrix_02">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="matrix_30">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="matrix_31">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QLabel" name="matrix_32">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QLabel" name="matrix_33">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="resetMatrix_Button">
             <property name="toolTip">
              <string>Resets the matrix to its initial state.</string>
             </property>
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Individual Transformation:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,1,1,1">
              <item>
               <widget class="QComboBox" name="transfo_ComboBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Angle, axis coordinates</string>
                </property>
                <property name="currentIndex">
                 <number>1</number>
                </property>
                <item>
                 <property name="text">
                  <string extracomment="Angle, axis coordinates">Rotation: </string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Translation: </string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Scaling:</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Normalize</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditA">
                <property name="toolTip">
                 <string>Angle value in degrees.</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>A</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditX">
                <property name="toolTip">
                 <string>Value along the x-axis.</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>X</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditY">
                <property name="toolTip">
                 <string>Value along the y-axis.</string>
                </property>
                <property name="placeholderText">
                 <string>Y</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="lineEditZ">
                <property name="toolTip">
                 <string>Value along the z-axis.</string>
                </property>
                <property name="placeholderText">
                 <string>Z</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0,0">
              <item>
               <widget class="QPushButton" name="undoButton">
                <property name="toolTip">
                 <string>Restores the previous state once.</string>
                </property>
                <property name="text">
                 <string>Undo</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="applyTransfo_Button">
                <property name="toolTip">
                 <string>Saves the current state and applies the transformation.</string>
                </property>
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="clearButton">
                <property name="toolTip">
                 <string>Clears the fields.</string>
                </property>
                <property name="text">
                 <string>Clear</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="validateLayout">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="validatePushButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Validate</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
